src/Parser.js

```diff
@@ -140,17 +140,18 @@ export default class Parser {
      * and fetches the one after that as the new look ahead.
      */
     consume() {
-        this.nextToken = this.gullet.get(false);
+        this.nextToken = this.gullet.get();
     }
 
     /**
-     * Switches between "text" and "math" modes, reconsuming nextToken
-     * in case it would be read differently in the new mode.
+     * Switches between "text" and "math" modes.
+     * This used to reconsume nextToken in the new mode because whitespace
+     * handling differed between the two nodes; now it just updates this.mode.
      */
     switchMode(newMode) {
-        this.gullet.unget(this.nextToken);
+        //this.gullet.unget(this.nextToken);
         this.mode = newMode;
-        this.consume();
+        //this.consume();
```

Comment:
remove commented out code

Reply:
I'm curious though why the `unget` and `consume` are no longer necessary.

Reply:
It used to be that `get` got passed true or false according to whether math mode was on, i.e., it ate spaces if it was in math mode.  But that's no longer the case (having moved that into parser, where it belongs according to TeX behavior), so now get doesn't depend on whether math mode is on, so there's no effect to ungetting and regetting.